flexmeasures/data/migrations/versions/524800c11eec_add_warning_and_error_as_annotations_.py

```diff
@@ -0,0 +1,24 @@
+"""add warning and error as annotations types
+
+Revision ID: 524800c11eec
+Revises: 4b5aa7856932
+Create Date: 2024-08-19 15:10:24.323594
+
+"""
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "524800c11eec"
+down_revision = "4b5aa7856932"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.execute("ALTER TYPE annotation_type ADD VALUE 'warning'")
+    op.execute("ALTER TYPE annotation_type ADD VALUE 'error'")
+
+
+def downgrade():
+    # Enum values cannot be removed in Postgres, so no downgrade action is provided
```

Comment:
Please add a reference.

And if this is the case, I suspect that upgrading, then downgrading and then upgrading again, will fail. To remedy that, perhaps we should only add the enum values in the upgrade command if they don't exist already.

Reply:
Good idea.

Reply:
I found a way to make downgrade work, but one little problem. That's when I comment in the last the line then `annotation_type` is empty. 
```python
def downgrade():
    op.execute("ALTER TYPE annotation_type RENAME TO annotation_type_old")

    op.execute("CREATE TYPE annotation_type AS ENUM('alert', 'holiday', 'label', 'feedback')")

    op.execute("""
            ALTER TABLE annotation
            ALTER COLUMN type 
            TYPE annotation_type
            USING type::text::annotation_type
        """)
    # op.execute("DROP TYPE annotation_type_old")
```

Reply:
I don't know the solution to this problem. I can only suggest to make sure that the upgrade command does not fail if the new enum values already exist. Did that fail in the first place?

Reply:
This looked like it had some more ideas: https://python-code.dev/articles/70502423

Reply:
Yes, it failed when I updated for the second time.

Reply:
it's quite complicated to drop the type because if it has data then that causes issues.